homeassistant/config_entries.py

```diff
@@ -850,7 +871,7 @@ def async_get_options_flow(config_entry: ConfigEntry) -> "OptionsFlow":
 
     @callback
     def _abort_if_unique_id_configured(
-        self, updates: Optional[Dict[Any, Any]] = None
+        self, updates: Optional[Dict[Any, Any]] = None, reload_on_update: bool = False,
```

Comment:
Looked the current use cases in our codebase, they can all reload. I suggest on making this the default behavior, as the would instantly improve all current cases.

```suggestion
        self, updates: Optional[Dict[Any, Any]] = None, reload_on_update: bool = True,
```

Reply:
Good call.

Reply:
I'm not sure Samsung is safe to reload

Reply:
@OnFreund If that is true, it is already a bug, as disabling an entity from Samsung would do the same.

Reply:
Samsung doesn't implement `async_unload_entry` and when you remove an entry from the UI it warns you that you need to restart HA.

Reply:
In that case, it will not reload @OnFreund, so nothing breaks.

Reply:
We should fix `samsung` here as that is a pretty poor UX.

I'll add a check for this case.

Reply:
Actually what i just added is overkill since we have

```
        supports_unload = hasattr(component, "async_unload_entry")

        if not supports_unload:
            if integration.domain == self.domain:
                self.state = ENTRY_STATE_FAILED_UNLOAD
            return False
```

I'll slice it down

Reply:
OK. Just hoping none of the other 30 or so integrations that use this don't break on reload...

Reply:
Setting it to `true` works in practice, but it breaks all the tests because they now get a reload where `async_setup_entry` is not mocked out.

Reply:
I don't think it will be that many to mock out though so probably doable to adjust the tests